external_search.py

```diff
@@ -17,12 +17,25 @@ class ExternalSearchIndex(object):
     work_document_type = 'work-type'
     __client = None
 
+    CURRENT_ALIAS_SUFFIX = '-current'
+    VERSION_RE = re.compile('-v([0-9]+)$')
+
+    @classmethod
+    def reset(cls):
+        """Resets the __client object to None so a new configuration
+        can be applied during object initialization.
+
+        This method is only intended for use in testing.
+        """
+        cls.__client = None
+
     def __init__(self, url=None, works_index=None):
     
         self.log = logging.getLogger("External search index")
 
         # By default, assume that there is no search index.
         self.works_index = None
+        from_configuration = False
```

Comment:
I think if you point to `works-v2` which doesn't exist yet and the alias is set to something else, that means you are setting up a new index. The code you're running with this `works-v2` config should use the index instead of the alias, and any other code that's still running somewhere else with your old config can keep using the alias and continue to work.

When you're ready, you should do a separate step to move the alias, and the other code that's running will automatically start using the new index without any config change or downtime.